modules/delegates.js

```diff
@@ -181,35 +181,26 @@ __private.validateBlockSlot = function(block, source, cb) {
 };
 
 /**
- * Gets slot time and keypair.
+ * Gets the assigned delegate to current slot and returns its keypair if present.
  *
  * @private
  * @param {number} slot
- * @param {number} height
+ * @param {number} round
  * @param {function} cb - Callback function
  * @returns {setImmediateCallback} cb, err, {time, keypair}
  * @todo Add description for the params
  */
-__private.getBlockSlotData = function(slot, height, cb) {
-	const round = slots.calcRound(height);
+__private.getDelegateKeypairForCurrentSlot = function(currentSlot, round, cb) {
```

Comment:
Just a thought as we are refactoring this function a lot, should we convert it to a Promise.

Reply:
As Promises are not heavily used across our application, I reckon it should be a separated task to the whole application.